### PR TITLE
test: Add coverage for bootstrap/Sentry

### PR DESCRIPTION
**Pull request by dcramer**
This also pulls in mocks for both xhr and fetch, as well as sentry-testkit.

### CODE REVIEW

**Comment by dcramer** [comment]
cc @zivl

File: tests/js/setup.js [hunk 1]
```diff
@@ -100,24 +101,19 @@ jest.mock('echarts-for-react/lib/core', () => {
 
 jest.mock('@sentry/browser', () => {
```
Comment (dcramer) [inline tests/js/setup.js]: @zivl maybe a pattern we should document

File: tests/js/spec/bootstrap.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        release: '123',
+      },
+    };
+
+    require('app');
+
+    const eventId = await Sentry.captureMessage('test');
+    expect(eventId).toBeDefined();
+    expect(Sentry.testkit.reports()).toHaveLength(1);
+    const {originalReport} = Sentry.testkit.reports()[0];
```
Comment (dcramer) [inline tests/js/spec/bootstrap.spec.jsx]: had to do this as sentry-testkit coerces the json payload into a 'report', but that report is missing some attributes (like release)

Comment (zivl) [reply tests/js/spec/bootstrap.spec.jsx]: @dcramer it is true we're omitting some attributes. Specifically, `release`,`user` we do pass over:
https://github.com/wix/sentry-testkit/blob/master/src/index.js#L29
I do agree though that `environment` should be part of  `SentryTestkit.Report`

File: tests/js/setup.js (hunk 1 above)
Comment (zivl) [reply tests/js/setup.js]: yeah this actually makes sense!
I've opened an issue at our end to put in the docs as well for suggested example...
https://github.com/wix/sentry-testkit/issues/44
maybe even predefined-ready-to-use-jest-mock? not sure but we'll check out that path

**Comment by dcramer** [comment]
This will be tough to get merged right now - specifically the Sentry change to sentry-testkit. Theres a bunch of spots that look for mocks, and they need both updated to use testkit, and ideally updated to use xhrmock. I'll slowly get this passing.

**Comment by BYK** [comment]
Closing due to staleness and conflicts.

**Pull request closed without merging**